external-crates/move/crates/move-compiler/src/linters/constant_naming.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! `ConstantNamingVisitor` enforces a naming convention for constants in Move programs,
+//! requiring them to follow an ALL_CAPS_SNAKE_CASE format. This lint checks each constant's name
+//! within a module against this convention.
+use crate::{
+    diag,
+    diagnostics::{
+        codes::{custom, DiagnosticInfo, Severity},
+        WarningFilters,
+    },
+    shared::{program_info::TypingProgramInfo, CompilationEnv},
+    typing::{
+        ast as T,
+        visitor::{TypingVisitorConstructor, TypingVisitorContext},
+    },
+};
+use move_ir_types::location::Loc;
+use move_symbol_pool::Symbol;
+
+use super::{LinterDiagCategory, LINTER_DEFAULT_DIAG_CODE, LINT_WARNING_PREFIX};
+
+/// Diagnostic information for constant naming violations.
+const CONSTANT_NAMING_DIAG: DiagnosticInfo = custom(
+    LINT_WARNING_PREFIX,
+    Severity::Warning,
+    LinterDiagCategory::ConstantNaming as u8,
+    LINTER_DEFAULT_DIAG_CODE,
+    "Constant name should be in all caps, snake case, pascal case or upper camel case.",
```

Comment:
```suggestion
    "non-conventional constant name",
```
The message on the error code should be relatively short

Reply:
"constant should follow naming convention" ?

Reply:
I like that a lot more, thanks!

Reply:
refactored